### PR TITLE
Implement password related functions

### PR DESCRIPTION
**Pull request by irenenguyen1017**
I've successfully implemented the forgot password functionality.

For local testing, please add the `EMAIL_USER` and `EMAIL_PASSWORD` values to your `.env.development file`. These credentials will be shared through a separate channel.

To test, register a new user on `Postman` using a `real email address`, and then initiate a password reset request. You'll receive an email from our shared account containing a URL. This URL leads to the password reset page, which is yet to be implemented.

Screenshot 2023-12-08 at 7 52 44 pm

Screenshot 2023-12-08 at 8 15 11 pm

The bug related to the user role during registration has been fixed.

Screenshot 2023-12-08 at 7 51 26 pm

Screenshot 2023-12-08 at 7 52 02 pm

Screenshot 2023-12-08 at 7 52 12 pm

### CODE REVIEW

File: src/controllers/AuthController.js [hunk 1]
```diff
@@ -80,3 +83,64 @@ export const login = async (req, res) => {
...
+      }
+    });
+
+    // Create the reset password URL, which includes the reset password token.
+    const resetUrl = `${envConfig.clientHost}/reset-password?token=${resetToken}`;
+
+    // Send an email to the user that includes the reset password URL
+    await transporter.sendMail({
+      from: {
+        name: 'Sakura pantry',
```
Comment (ellenpham) [inline src/controllers/AuthController.js:131]: Just a minor detail, but can you change this to Sakura Pantry 😄

File: src/routes/AuthRoutes.js [hunk 2]
```diff
@@ -1,9 +1,11 @@
 import express from 'express';
-import { register, login } from '../controllers/AuthController.js';
+import { register, login, forgotPassword } from '../controllers/AuthController.js';
 
 const router = express.Router();
 
 router.post('/auth/register', register);
```
Comment (ellenpham) [inline src/routes/AuthRoutes.js:6]: I was thinking, the authRoutes are currently look like this ` localhost:5000/api/auth/register` or `localhost:5000/api/auth/login`. 

Maybe no need `api` in the routes for the authRoutes?

File: src/controllers/AuthController.js [hunk 3]
```diff
@@ -80,3 +83,64 @@ export const login = async (req, res) => {
...
+        address: envConfig.mail.user
+      },
+      to: existingUser.email,
+      subject: 'Password Reset',
+      html: `<p>You requested a password reset. Please go to this link to reset your password:<p>\
```
Comment (ellenpham) [inline src/controllers/AuthController.js:136]: Maybe you can make the email content a bit longer and add information about the link expiration. Something like this:

image

**Review by ellenpham — CHANGES_REQUESTED** [review]

File: src/routes/AuthRoutes.js (hunk 2 above)
Comment (irenenguyen1017) [reply src/routes/AuthRoutes.js:6]: I thought all the routes for the back-end started from the api, didn't they?

After a quick search on ChatGPT, do you still want to remove api from the route?

Screenshot 2023-12-08 at 9 56 26 pm

Comment (ellenpham) [reply src/routes/AuthRoutes.js:6]: all good, we def can keep the `api`.

File: src/controllers/AuthController.js (hunk 1 above)
Comment (irenenguyen1017) [reply src/controllers/AuthController.js:131]: Okay😆

**Pull request merged**